src/asset.ts

```diff
@@ -11,28 +11,52 @@ class Asset {
   public ledger: Ledger;
   public name: string;
   public contract?: string;
+  public decimals?: number;
 
-  public constructor(name: string, ledger: Ledger, contract?: string) {
+  public constructor(
+    name: string,
+    ledger: Ledger,
+    contract?: string,
+    decimals?: number
+  ) {
     this.name = name;
     this.ledger = ledger;
     this.contract = contract;
+    this.decimals = decimals;
   }
 }
+export function toAssetFromName(name: string): Asset | undefined {
+  return toAsset({ name });
+}
 
-export function toAsset(asset: string): Asset | undefined {
-  switch (asset) {
+export function toAsset(
+  asset: any,
+  ledger?: Ledger,
```

Comment:
🔧 Seems like you don't use `ledger` besides within the function `tokensCreateAsset`. Imho you could then omit it here?

Reply:
Interesting point, I'll see if I can optimise it.

Reply:
I have added a TODO because this `tokensCreateAsset` is 1:1 with `Tokens.createAsset`.
As I only put the glue code in the next PR, I don't want to do the change to then realise I simply cannot glue it together.

Once everything is in place I'll hopefully be able to remove the TODOs and do this optimisation.